client/src/components/TimeZoneDropdown.js

```diff
@@ -0,0 +1,174 @@
+    },
+    {
+        key: '-1000',
+        value: '-1000',
+        text: 'HST/SDT (UTC/GMT -10)',
+    },
+    {
+        key: '-900',
+        value: '-900',
+        text: 'AKST/HDT (UTC/GMT -9)',
+    },
+    {
+        key: '-800',
+        value: '-800',
+        text: 'PST/AKDT (UTC/GMT -8)',
+    },
+    {
+        key: '-700',
+        value: '-700',
+        text: 'MST/PDT (UTC/GMT -7)',
+    },
+    {
+        key: '-600',
+        value: '-600',
+        text: 'CST/MDT (UTC/GMT -6)',
+    },
+    {
+        key: '-500',
+        value: '-500',
+        text: 'EST/CDT (UTC/GMT -5)',
+    },
+    {
+        key: '-400',
+        value: '-400',
+        text: 'AST/EDT (UTC/GMT -4)',
+    },
+    {
+        key: '-300',
+        value: '-300',
+        text: 'ART/ADT (UTC/GMT -3)',
+    },
+    {
+        key: '-200',
+        value: '-200',
+        text: 'BST (UTC/GMT -2)',
+    },
+    {
+        key: '-100',
+        value: '-100',
+        text: 'CVT/EGT (UTC/GMT -1)',
+    },
+    {
+        key: '0',
+        value: '0',
+        text: 'UTC/GMT +0',
+    },
+    {
+        key: '100',
+        value: '100',
+        text: 'BST/IST/WAT (UTC/GMT +1)',
+    },
+    {
+        key: '200',
+        value: '200',
+        text: 'CAT/EET (UTC/GMT +2)',
+    },
+    {
+        key: '300',
+        value: '300',
+        text: 'EAT/FET (UTC/GMT +3)',
+    },
+    {
+        key: '400',
+        value: '400',
+        text: 'AZT/GST (UTC/GMT +4)',
+    },
+    {
+        key: '500',
+        value: '500',
+        text: 'PKT/TJT (UTC/GMT +5)',
+    },
+    {
+        key: '600',
+        value: '600',
+        text: 'BST/KGT (UTC/GMT +6)',
+    },
+    {
+        key: '700',
+        value: '700',
+        text: 'ICT/WIT (UTC/GMT +7)',
+    },
+    {
+        key: '800',
+        value: '800',
+        text: 'AWST/CT/HKT (UTC/GMT +8)',
+    },
+    {
+        key: '900',
+        value: '900',
+        text: 'EIT/JST (UTC/GMT +9)',
+    },
+    {
+        key: '1000',
+        value: '1000',
+        text: 'AEST/PGT (UTC/GMT +10)',
+    },
+    {
+        key: '1100',
+        value: '1100',
+        text: 'AEDT/NFT (UTC/GMT +11)',
+    },
+    {
+        key: '1200',
+        value: '1200',
+        text: 'NZST (UTC/GMT +12)',
+    }
+]
+
+export default class TimeZoneDropdown extends Component {
+    state={
+        offset: 0
+    }
+
+    componentWillMount() {
+        this.setState({offset: this.props.userDetails.timeZone.toString()})
+    }
+
+    async updateTimeZone(value) {
+        try {
+            let update = await makeCall({
+                id: this.props.userDetails._id,
+                role: this.props.userRole,
+                timeZone: parseInt(value)
+            }, '/changeTimeZone', 'patch')
+        } catch (e) {
+            console.log('timezone update failed')
+        }
+    }
+
+    handleDropdownChange = (e, { name, value }) => {
+        this.setState({ [name]: value })
```

Comment:
Can you try 
```
this.setState({ ... }, () => {
this.updateTimeZone(value);
this.props.liftTimezone(value);
})
```
and see if you have any luck with timezone updating as expected? Then again you aren't using `this.state.value` as args for the methods that execute after the async setState

Reply:
The issue with timezone updating is actually from refreshProfile and the way that components receive updated props - just like I had to use componentWillReceiveProps in schedulings and requests, I would need something similar in each component that receives userDetails.
ComponentDidUpdate would be a decent candidate for this, but seems to run as frequently as it possibly can which is bad for performance. componentWillReceiveProps works pretty well for schedulings and requests but seemed to be unreliable for the dropdown possibly because it requires a complete re-render